orc8r/gateway/python/magma/magmad/metrics_collector.py

```diff
@@ -93,8 +98,10 @@ def sync(self):
                 # If we throw an exception here, we'll have no idea whether
                 # something was postprocessed or not, so I guess try and make it
                 # idempotent?  #m sevchicken
-                self.post_processing_fn(self._samples)
-            samples = self._retry_queue + self._samples
+                self.post_processing_fn(self._samples_for_service[service_name])
+            samples = self._retry_queue_for_service[service_name] + \
```

Comment:
Given that every service might have non uniform number of metrics, It would be good to do some batching of metrics per service name as well.